kernel/cooking.ml

```diff
@@ -183,18 +184,13 @@ let lift_univs cb subst auctx0 =
     assert (AUContext.is_empty auctx0);
     subst, (Monomorphic_const ctx)
   | Polymorphic_const auctx ->
-    if (Univ.LMap.is_empty subst) then
+    if (Univ.Instance.is_empty subst) then
+      (** Still need to take the union for the constraints between globals *)
       subst, (Polymorphic_const (AUContext.union auctx0 auctx))
     else
-      let len = Univ.LMap.cardinal subst in
-      let rec gen_subst i acc =
-        if i < 0 then acc
-        else
-          let acc = Univ.LMap.add (Level.var i) (Level.var (i + len)) acc in
-          gen_subst (pred i) acc
-      in
-      let subst = gen_subst (Univ.AUContext.size auctx - 1) subst in
-      let auctx' = Univ.subst_univs_level_abstract_universe_context subst auctx in
+      let ainst = Univ.make_abstract_instance auctx in
+      let subst = Instance.append subst ainst in
+      let auctx' = Univ.subst_univs_level_abstract_universe_context (Univ.make_instance_subst subst) auctx in
```

Comment:
I can't figure out what this part of the code is supposed to do so I don't know if it's correct.
I would say if travis passes it's probably correct but @ppedrot's travis has imploded with apt-get failing.

Reply:
This code is essentially trying to perform a quantification over an abstract context prefix. It's like a lift  function for universes, mostly.

Reply:
@ppedrot Please add a comment for mortals like me, thanks!